js/localization/ldml/date.parser.js

```diff
@@ -218,6 +219,22 @@ export const getRegExpInfo = function(format, dateParts) {
 
     addPreviousStub();
 
+    patterns.every((pattern, index) => {
```

Comment:
Actually, it's better to extract this code to the separate function. The `getRegExpInfo` method becomes too big and it's hard to read the code.